solr/core/src/java/org/apache/solr/handler/component/SearchHandler.java

```diff
@@ -355,18 +356,18 @@ public void handleRequestBody(SolrQueryRequest req, SolrQueryResponse rsp) throw
     final RTimerTree timer = rb.isDebug() ? req.getRequestTimer() : null;
 
     final CircuitBreakerRegistry circuitBreakerRegistry = req.getCore().getCircuitBreakerRegistry();
-    if (circuitBreakerRegistry.isEnabled()) {
+    if (circuitBreakerRegistry.isEnabled(RequestType.SEARCH)) {
```

Comment:
Merge conflict with #1882 here, sorry about that. Is it okay if I go ahead and resolve it?